### PR TITLE
fix(Table): fix desc / asc order

### PR DESCRIPTION
**Pull request by spartDev**
- [ ] Feature
- [x] Fix
- [ ] Enhancement

## Brief
- Fix desc / asc order in `<Table />`
- Remove `lodash` dependency  to have a smaller build size.

## Description
There was a bug that did not allow the `<Table />` to be ordering by `desc`.
I remove completely `lodash` library, I think is overwhelming for just ordering an array and we gain a smaller build size.

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/helpers/sortElementsInArray.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * If the return number is positive, B will be shown before A.
+ * If the return number is 0, A and B will remain in the same order as when they entered the loop.
+ *
+ * @param {string, number} a
+ * @param {string, number} b
+ *
+ * @return {number}
+ */
+
+export default function sortElementsInArray(a, b) {
```
Comment (leopoldhoudin) [inline src/helpers/sortElementsInArray.js]: Wow... you could use ternary expressions for this!

```javascript

return b > a ? 1 : b < a ? -1 : 0;

```

This why the documentation is important!

File: src/Table/index.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
-import SortBy from 'lodash/sortBy';
-import Reverse from 'lodash/reverse';
 
 import { Icon, Theme } from '..';
+import sortElementsInArray from '../helpers/sortElementsInArray';
```
[the remark this answers is not in the document]
Comment (leopoldhoudin) [reply src/Table/index.js]: Well, this naming does not provide the above given explanation... so... not sure either of these names are appropriate.

Furthermore, the function does not sort. So I don't get why it should be called `sort---whatever`. It compares... Hence the following:

In most programming languages, the function you described above is called `compare`.

*Reference*:
- [Java 8/9](https://docs.oracle.com/javase/9/docs/api/java/lang/String.html#compareTo-java.lang.String-)
- [.NET](https://docs.microsoft.com/en-us/dotnet/api/system.collections.icomparer.compare?view=netframework-4.7.2)
- [Javascript local](https://stackoverflow.com/questions/2167602/optimum-way-to-compare-strings-in-javascript)

File: src/Table/index.js [hunk 3]
```diff
@@ -195,11 +194,16 @@ class Table extends PureComponent {
     });
     if (index >= 0) {
       if (direction === 'asc') {
-        sortedData = SortBy(sortedData, [d => colsDef[index].value(d.item)]);
+        sortedData = sortedData.sort(
```
Comment (leopoldhoudin) [inline src/Table/index.js]: You refactor the two sorting blocks:

```javascript

if (index >= 0) {
  sortedData = sortedData.sort(
    (a, b) =>
      (direction === 'asc' ? 1 : -1) compare(colsDef[index].value(a.item), colsDef[index].value(b.item))
  );
}

```

File: src/helpers/__tests__/sortElementsInArray.test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import sortElementsInArray from '../sortElementsInArray';
+
+describe('sortElementsInArray helpers', () => {
+  it('should return 0', () => {
+    const expectedValue = sortElementsInArray(1, 1);
+    expect(expectedValue).toMatchSnapshot();
```
Comment (leopoldhoudin) [inline src/helpers/__tests__/sortElementsInArray.test.js]: lol, do you really need a snapshot to check that the value is `0`???

File: src/helpers/__tests__/sortElementsInArray.test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  });
+  it('should return -1', () => {
+    const expectedValue = sortElementsInArray(2, 1);
+    expect(expectedValue).toMatchSnapshot();
```
Comment (leopoldhoudin) [inline src/helpers/__tests__/sortElementsInArray.test.js]: lol, do you really need a snapshot to check that the value is `-1`???

File: src/helpers/__tests__/sortElementsInArray.test.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  });
+  it('should return 1', () => {
+    const expectedValue = sortElementsInArray(1, 2);
+    expect(expectedValue).toMatchSnapshot();
```
Comment (leopoldhoudin) [inline src/helpers/__tests__/sortElementsInArray.test.js]: lol, do you really need a snapshot to check that the value is `1`???

File: src/Table/index.js (hunk 3 above)
Comment (spartDev) [reply src/Table/index.js]: @leopoldhoudin this can not work

File: src/helpers/__tests__/compare.test.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import compare from '../compare';
+
+describe('sortElementsInArray helpers', () => {
```
Comment (AntoninMarchardDev) [inline src/helpers/__tests__/compare.test.js]: Change it for the new function / file name

File: src/Table/index.js (hunk 3 above)
Comment (spartDev) [reply src/Table/index.js]: @leopoldhoudin fix

**Pull request merged**